resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -558,17 +567,24 @@ public class ApiClient implements AutoCloseable {
 
     private <T> ApiResponse<T> getAPIResponse(ApiRequest<?> request, TypeReference<T> returnType, boolean isAuthRequest) throws IOException, ApiException {
         ApiClientConnectorRequest connectorRequest = prepareConnectorRequest(request, isAuthRequest);
+        Retry retry = new Retry();
         ApiClientConnectorResponse connectorResponse = null;
+        long timeElapsedPerRequestInMills = 0L;
         try {
-            connectorResponse = connector.invoke(connectorRequest);
+            do {
+                Stopwatch timeElapsedPerRequest = Stopwatch.createStarted();
```

Comment:
Don't track the time of individual requests. Create the stopwatch in the Retry constructor and store it in Retry. Then use the elapsedTime method on the stopwatch to check how long it's been since the process started. No updating, no math, no logic except a check against the elapsed time.

Reply:
By doing this it will also include the time spent for the first 5 retries for 50x before coming to 5 minutes buckets (intermediate 429) while calculating 5 minutes buckets time. If we ignore that we can use one stopwatch across the process.

Reply:
Yes, that's what I'm asking you to do. Measure the **total** time since the first request was made. All you need to do is measure if stopwatch.elapsed() < 5 minutes, < 10 minutes, or else to determine which value to use for the delay. This might even work with a single expression using something like this (pseudocode, won't compile): `double delay = Math.pow(duration, Math.floor((stopwatch.elapsed(seconds) / (60 * 5)) + 1);`.